src/main/java/uk/ac/ebi/atlas/experimentpage/ExternallyAvailableContentController.java

```diff
@@ -0,0 +1,86 @@
+package uk.ac.ebi.atlas.experimentpage;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+import org.springframework.http.MediaType;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseBody;
+import uk.ac.ebi.atlas.model.download.ExternallyAvailableContent;
+
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.text.MessageFormat;
+import java.util.List;
+
+import static org.apache.commons.lang3.StringUtils.isNotEmpty;
+import static uk.ac.ebi.atlas.utils.GsonProvider.GSON;
+
+@Controller
+public class ExternallyAvailableContentController {
+    private final ExpressionAtlasContentService expressionAtlasContentService;
+
+    public ExternallyAvailableContentController(ExpressionAtlasContentService expressionAtlasContentService) {
+        this.expressionAtlasContentService = expressionAtlasContentService;
+    }
+
+    /*
+    I could be nicer and maybe even have tests:
+    - the "View in Array Express" needs an independent URL
+    - the QC report needs a redirect to another Atlas page, preserving the access key
+    - the typical resource needs to circle back to this page
+    */
+    private JsonObject contentAsJson(ExternallyAvailableContent content, String accession, String accessKey) {
```

Comment:
Would it be better if we could move private methods to the bottom of the class? and public on the top of the class.